api/api.js

```diff
@@ -0,0 +1,165 @@
+import axios from "axios";
+import { TOKEN } from "@/constants";
+const { ACCESS_TOKEN, REFRESH_TOKEN } = TOKEN;
+export const instance = axios.create({
+  baseURL: process.env.NEXT_PUBLIC_API_HOST
+});
+instance.interceptors.request.use(
+  (config) => {
+    // 토큰이 필요 없는 요청의 URL을 확인
+    const noAuthRequiredEndpoints = [
+      "/auth/signIn",
+      "/auth/signUp",
+      "/products",
+      "/articles"
+    ];
+    // 요청 URL이 noAuthRequiredEndpoints에 포함되어 있지 않을 경우
+    if (!noAuthRequiredEndpoints.includes(config.url)) {
+      const accessToken = localStorage.getItem(ACCESS_TOKEN);
+      if (accessToken) {
+        config.headers.Authorization = `Bearer ${accessToken}`;
+      }
+    }
+    return config;
+  },
+  (error) => {
+    return Promise.reject(error);
+  }
+);
+
+instance.interceptors.response.use(
+  (response) => response,
+  async (error) => {
+    const originalRequest = error.config;
+    try {
+      if (error.response && error.response.status === 401) {
```

Comment:
@taeyeonkim94 이건 아래와 같이 쓸 수 있어요
```suggestion
      if (error?.response.status === 401) {
```